### PR TITLE
hevc: Remove need for 2nd buffer

### PR DESCRIPTION
**Pull request by papafelice**

### CODE REVIEW

**Comment by FernetMenta** [comment]
thanks!

the commit message of 6df191a seems not appropriate :)
could you please split the bitstream changes into a separate commit? then I can test it with the other parsers for regressions.

**Comment by papafelice** [comment]
Ok, I will try so. But the non-buffer version of hevc needs a different bitstream. So I will make an intermediate bsEP3 version that will later be reunited with the optimized bitstream. Ok?

**Comment by FernetMenta** [comment]
why don't you create a specialized version (subclass) of bitsream, that implements EP3?

**Comment by papafelice** [comment]
Am I right to assume that H.264 is used in DVB-S2? I have a DVB-S2 USB adapter lying around that I could hook to a dish and test the H.264 parser.

**Comment by FernetMenta** [comment]
> Am I right to assume that H.264 is used in DVB-S2?

Yes. Most HD channels are h264, SD are mpeg2

**Comment by papafelice** [comment]
Made it. Used my wife's PC as VDR server, hooked the DVBS-usb thingy to it. wirbelscan etc, Copied my latest patches to it

Various HD Channels in HD with h264 -> Works
Various HD Channels in HD with hevc -> Works
SES UHD Demo 4K hevc -> Works!  (Though I get pixel errors and my Laptop where Kodi runs on starts boiling due to the S/W decoding :-)

**Comment by papafelice** [comment]
Damn, clicked the wrong button

**Comment by FernetMenta** [comment]
Do you need help with restructuring the commits?

File: bitstream.h [hunk 1]
```diff
@@ -34,21 +34,25 @@ class cBitstream
   int      m_offset;
   int      m_len;
   bool     m_error;
+  bool     m_doEP3;
```
Comment (FernetMenta) [inline bitstream.h]: why did you not override cBitstream, i.e.
class cBitstreamEP3 : public cBitstream

Comment (papafelice) [inline bitstream.h]: That was my first though, but I wasn't sure if I don't override readGolombUE too, would a call to readGolombUE call the cBitstreamEP3::readBits  or cBitstream::readBits, so I duplicated the functions needed in order to leave original code untouched. 

Then, in the second commit, I wanted to do only the bitstream optimization as requested. But in any case, that requires a change in parse_hevc for the class name, so this change will also be in the commit.

**Comment by papafelice** [comment]
"...Do you need help with restructuring the commits?..."
Yes, please. If not for now, then for the future (see above: H264 and HEVC are doing fine up to 4K with the new bitstream version)

**Comment by FernetMenta** [comment]
there are some minor coding style issues left but I will clean them after merge. Enough git-foo for this time. Thanks very much!

**Pull request merged**